src/shared/components/query/QueryStore.ts

```diff
@@ -563,6 +504,194 @@ export class QueryStore
+
+			let result:{[studyId:string]:VirtualStudy} = {}
+			
+			await Promise.all(unknownQueriedIds.map(id =>{
+				return new Promise((resolve, reject) => {
+					sessionServiceClient.getVirtualStudy(id).then((virtualStudy)=>{
+						//physical study ids iin virtual study
+						let ids = virtualStudy.data.studies.map(study=>study.id);
+						//unknown/unauthorized studies within virtual study
+						let unKnownPhysicalStudyIds = ids.filter(id => !physicalStudiesIdsSet[id]);
+						if(_.isEmpty(unKnownPhysicalStudyIds)){
+							result[id] = virtualStudy;
+						}
+						resolve();
+					}).catch(() => {//error is thrown when the id is not found
+						resolve();
+					});
+				});
+			}));
+			return result;
+		},
+		default: {}
+	});
+
+	//this is mainly used when custom case set is selected
+	private readonly selectedStudyToSampleSet = remoteData<{[id:string]:{[id:string]:boolean}}>({
+		await: ()=>[this.userVirtualStudiesSet, this.sharedVirtualStudiesSet],
+		invoke: async ()=>{
+			let studyToSampleSet:{[id:string]:{[id:string]:boolean}} = {};
+
+			const _vs = {...this.userVirtualStudiesSet.result, ...this.sharedVirtualStudiesSet.result};
+
+			const physicalStudyIds = _.filter(this._allSelectedStudyIds.keys(),studyId=> _vs[studyId] === undefined);
+
+			if(this._allSelectedStudyIds.keys().length !== physicalStudyIds.length){
+				await Promise.all(_.map(physicalStudyIds, studyId => {
+					return client.getAllSamplesInStudyUsingGET({
+						studyId: studyId
+					}).then(samples => {
+						studyToSampleSet[studyId] = stringListToSet(samples.map(sample => sample.sampleId));
+					})
+				}));
+	
+				this._allSelectedStudyIds.keys().forEach(id=>{
+					if(_vs[id]){
+						let virtualStudy = _vs[id];
+						virtualStudy.data.studies.forEach(study => {
+							if(studyToSampleSet[study.id] === undefined){
+								studyToSampleSet[study.id] = stringListToSet(study.samples)
+							}
+						});
+					}
+				});
+			} else {
+				physicalStudyIds.forEach(studyId=>{
+					studyToSampleSet[studyId] = {}
+				})
+			}
+			return studyToSampleSet;
+		},
+		default: {}
+	})
+
+	private readonly selectableStudiesSet = remoteData<{[studyId:string]:string[]}>({
+		await: ()=>[this.physicalStudiesSet, this.selectedStudyToSampleSet],
+		invoke: async ()=>{
+			let result:{[studyId:string]:string[]} = {}
+		result = _.reduce(this.physicalStudiesSet.result,(acc, next, key)=>{
+			acc[key] = [key]
+			return acc;
+		},result)
+
+		result = _.reduce(this.userVirtualStudiesSet.result,(acc, next, key)=>{
+			acc[key] = [key]
+			return acc;
+		},result)
+
+		result = _.reduce(this.sharedVirtualStudiesSet.result,(acc, next, key)=>{
+			acc[key] = next.data.studies.map(study=> study.id)
+			return acc;
+		},result)
+		return  result;
+		},
+		default: {},
+		onResult:(selectableStudiesSet)=>{
+			if(this.selectedSampleListId !== CUSTOM_CASE_LIST_ID) {

```

Comment:
lets make a practice of always putting a comment where we use onResult explaining why we need to use onResult.   Basically, my understanding is that it's because we want to set a default user selection, right?